aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+        hand_cam_intrinsics = self._to_camera_metadata_dict(hand_cam_intrinsics)
+        hand_cam_pose_estimator = AprilTagPoseEstimator(hand_cam_intrinsics)
+
+        # Register marker ids
+        marker_ids_list = [DOCK_ID]
+        # marker_ids_list = [i for i in range(521, 550)]
+        hand_cam_pose_estimator.register_marker_ids(marker_ids_list)
+
+        marker_position_from_dock_list = []  # type: List[Any]
+        marker_quaternion_form_dock_list = []  # type: List[Any]
+
+        marker_position_from_robot_list = []  # type: List[Any]
+        marker_quaternion_form_robot_list = []  # type: List[Any]
+
+        while len(marker_position_from_dock_list) < data_size_for_avg:
+            print(f"Iterating - {len(marker_position_from_dock_list)}")
+            is_marker_detected_from_hand_cam = False
+            img_response_hand = self.spot.get_hand_image()
+            img_hand = image_response_to_cv2(img_response_hand)
+
+            (
+                img_rend_hand,
+                hand_sp_handcam_T_marker,
+            ) = hand_cam_pose_estimator.detect_markers_and_estimate_pose(
+                img_hand, should_render=True
+            )
+
+            if hand_sp_handcam_T_marker is not None:
+                print("Trackedddd")
+                is_marker_detected_from_hand_cam = True
+                hand_mn_handcam_T_marker = (
+                    Spot.convert_transformation_from_sophus_to_magnum(
+                        hand_sp_handcam_T_marker
+                    )
+                )
+
+            # Spot - spotWorld_T_handcam computation
+            frame_tree_snapshot_hand = img_response_hand.shot.transforms_snapshot
+            hand_mn_body_T_handcam = self._get_body_T_handcam(frame_tree_snapshot_hand)
+            hand_mn_spotWorld_T_handcam = self._get_spotWorld_T_handcam(
+                frame_tree_snapshot_hand, spot_frame=spot_frame
+            )
+
+            if is_marker_detected_from_hand_cam:
+                hand_mn_spotWorld_T_marker = (
+                    hand_mn_spotWorld_T_handcam @ hand_mn_handcam_T_marker
+                )
+
+                hand_mn_body_T_marker = (
+                    hand_mn_body_T_handcam @ hand_mn_handcam_T_marker
+                )
+
+                img_rend_hand = decorate_img_with_text(
+                    img=img_rend_hand,
+                    frame_name=spot_world_frame,
+                    position=hand_mn_spotWorld_T_marker.translation,
+                )
+
+                dist = hand_mn_handcam_T_marker.translation.length()
+
+                print(
+                    f"Dist = {dist}, Recordings - {len(marker_position_from_dock_list)}"
+                )
+                if dist < filter_dist:
+                    marker_position_from_dock_list.append(
+                        np.array(hand_mn_spotWorld_T_marker.translation)
+                    )
+                    marker_quaternion_form_dock_list.append(
```

Comment:
typo: `marker_quaternion_[form->from]_dock_list`

Reply:
addressed. thanks for finding this